promoter/promoter.go

```diff
@@ -223,14 +223,14 @@ func (p *Promoter) threadedPollTransactions() {
 		// For each one get the related txns from skyd and save them to
 		// the db.
 		var nAddresssInserted, nTxnsInserted int
-		for c.Next(p.staticBGCtx) {
-			// Get address.
-			var wa WatchedAddress
-			if err := c.Decode(&wa); err != nil {
-				p.staticLogger.WithError(err).Error("Failed to decode address")
-				continue // try next address
-			}
+		// Get addresses.
+		var was []WatchedAddress
+		if err := c.All(p.staticBGCtx, &was); err != nil {
+			p.staticLogger.WithError(err).Error("Failed to decode address")
+			continue // try next address
```

Comment:
nit: We only get one error for the whole set of addresses. The `continue` is fine but the comment and the error are misleading.